src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,420 @@
+namespace Lumper.UI.Updater;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text.Json.Nodes;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Lumper.Lib.BSP.IO;
+using Lumper.UI.Views;
+using MsBox.Avalonia;
+using MsBox.Avalonia.Enums;
+using Newtonsoft.Json;
+using NLog;
+
+internal sealed partial class Updater
+{
+    [GeneratedRegex(@"^(\d+)\.(\d+)\.(\d+)")]
+    private static partial Regex VersionRegex();
+    /// <summary>
+    /// record for deserializing JSON objects 
+    /// given in the GitHub JSON API response 
+    /// </summary>
+    private sealed record Asset
+    {
+        [JsonProperty("browser_download_url")]
+        public string BrowserDownloadUrl { get; set; }
+        [JsonProperty("name")]
+        public string Name { get; set; }
+    }
+    /// <summary>
+    ///  record for deserializing Github API response data
+    ///  see https://api.github.com/repos/momentum-mod/lumper/releases for the full format
+    /// </summary>
+
+    private sealed record GithubUpdate
+    {
+        [JsonProperty("tag_name")]
+        public string TagName { get; set; }
+        [JsonProperty("assets")]
+        public Asset[] Assets { get; set; }
+
+    }
+    /// <summary>
+    /// Major/Minor/Patch format
+    /// </summary>
+    /// <param name="major">First digits of the version format</param>
+    /// <param name="minor">Second digits of the version format</param>
+    /// <param name="patch">Third digits of the version format</param>
+    public sealed record Version(int major, int minor, int patch)
+    {
+        public int Major { get; } = major;
+        public int Minor { get; } = minor;
+        public int Patch { get; } = patch;
+        public override string ToString() => $"{major}.{minor}.{patch}";
+    }
+    /// <summary>
+    /// Parses a string to find the major/minor/patch versioning
+    /// </summary>
+    /// <param name="s">String containing a version number following the format 'xx.yy.zz'</param>
+    private static Version? GetVersionFromString(string s)
+    {
+        //match pattern of xx.yy.zz
+        Match match = VersionRegex().Match(s);
+        if (match.Success)
+        {
+            GroupCollection currentVersion = match.Groups;
+
+
+            int.TryParse(currentVersion[1].ToString(), out var major);
```

Comment:
`TryParse` here returns a boolean whether the parse was successful, which you're not using. `TryParse` has an evil sibling, `Parse`, which will throw if it fails. But as I've rambled about below, throwing for a parsing error like that is completely fine. In fact it'd be *very* weird if the parse failed here, given our regex was successful! So all you need is 

```cs
return new Version(int.Parse(currentVersion[1].Value, currentVersion[2].Value, currentVersion[3].Value))`
```
note I'm using `.Value` on the GroupCollection instead of ToString. If you peek at the source, `ToString` just returns `Value`, but `Value` is already a string and is what you should be accessing.